src/js/discharge-wizard/components/FormQuestions.jsx

```diff
@@ -0,0 +1,304 @@
+          value={{ value: dischargeYear }}
+          onValueChange={(update) => { this.updateField(key, update.value); }}/>
+      </fieldset>
+    );
+  }
+
+  renderQuestionTwoB() {
+    const key = '5_dischargeMonth';
+    if (!shouldShowQuestion(key, this.props.formValues.questions)) { return null; }
+
+    const monthLabel = (
+      <legend className="legend-label">
+        <h4>What month were you discharged?</h4>
+      </legend>
+    );
+
+    return (
+      <fieldset className="fieldset-input dischargeMonth" key="dischargeMonth">
+        <ErrorableSelect
+          autocomplete="false"
+          label={monthLabel}
+          name={key}
+          options={months}
+          ref={(el) => { this[key] = el; }}
+          value={{ value: this.props.formValues[key] }}
+          onValueChange={(update) => { this.updateField(key, update.value); }}/>
+      </fieldset>
+    );
+  }
+
+  renderQuestionThree() {
+    const key = '6_courtMartial';
+    if (!shouldShowQuestion(key, this.props.formValues.questions)) { return null; }
+
+    const label = <h4>Was your discharge the outcome of a General Court Martial? (Answer “no” if your discharge was administrative, or was the outcome of a Special or a Summary Court Martial.)</h4>;
+    const options = [
+      { label: 'Yes', value: '1' },
+      { label: 'No', value: '2' },
+    ];
+
+    return this.renderQuestion(key, label, options);
+  }
+
+  renderQuestionFour() {
+    const key = '7_branchOfService';
+    if (!shouldShowQuestion(key, this.props.formValues.questions)) { return null; }
+
+    const label = <h4>In which branch of service did you serve?</h4>;
+    const options = [
+      { label: 'Army', value: 'army' },
+      { label: 'Navy', value: 'navy' },
+      { label: 'Air Force', value: 'airForce' },
+      { label: 'Coast Guard', value: 'coastGuard' },
+      { label: 'Marines', value: 'marines' },
+    ];
+
+    return this.renderQuestion(key, label, options);
+  }
+
+  renderQuestionFive() {
+    const key = '8_prevApplication';
+    if (!shouldShowQuestion(key, this.props.formValues.questions)) { return null; }
+
+    const label = <h4>Have you previously applied for a discharge upgrade for this period of service?</h4>;
+    const options = [
+      { label: 'Yes', value: '1' },
+      { label: 'No', value: '2' },
+    ];
+
+    return this.renderQuestion('8_prevApplication', label, options);
+  }
+
+  renderQuestionFiveA() {
+    const key = '9_prevApplicationYear';
+    if (!shouldShowQuestion(key, this.props.formValues.questions)) { return null; }
+
+    const prevApplicationYearLabel = <h4>What year did you make this application?</h4>;
+    let labelYear;
+
+    switch (this.props.formValues['8_prevApplication']) {
+      case '1':
+      case '2':
+        labelYear = 2014;
+        break;
+      case '3':
+        labelYear = 2011;
+        break;
+      case '4':
+        labelYear = 2017;
+        break;
+      default:
+        break;
+    }
```

Comment:
Would it possibly be better to move these into a map? Like:

`const labelYears = { '1': 2014, '2': 2014, '3'...`

Reply:
yes it would. 100% want to refactor this entire app and clean things up once the dust settles